lib/SimpleHttpClient/Communicator.cpp

```diff
@@ -445,12 +445,10 @@ void Communicator::handleResult(CURL* handle, CURLcode rc) {
     (*rip->_options._curlRcFn)(rc);
   }
 
+  MUTEX_LOCKER(guard, _handlesLock);
   auto inProgress = _handlesInProgress.find(rip->_ticketId);
   auto curlHandle = (*inProgress).second->getSharedPtr();
-  {
-    MUTEX_LOCKER(guard, _handlesLock);
-    _handlesInProgress.erase(rip->_ticketId);
-  }
+  _handlesInProgress.erase(rip->_ticketId);
```

Comment:
We could erase here by using `inProgress` instead of using `rip->_ticketId`. The former is probably slightly faster than the latter.